src/herbpy/herb.py

```diff
@@ -30,17 +31,38 @@ def initialize(robot_xml=None, env_path=None, attach_viewer=False,
         if not env.Load(env_path):
             raise Exception('Unable to load environment frompath %s' % env_path)
 
+    herb_name = None
     # Load the URDF file into OpenRAVE.
     urdf_module = RaveCreateModule(env, 'urdf')
     if urdf_module is None:
         logger.error('Unable to load or_urdf module. Do you have or_urdf'
                      ' built and installed in one of your Catkin workspaces?')
         raise ValueError('Unable to load or_urdf plugin.')
 
-    urdf_uri = 'package://herb_description/robots/herb.urdf'
-    srdf_uri = 'package://herb_description/robots/herb.srdf'
-    args = 'Load {:s} {:s}'.format(urdf_uri, srdf_uri)
-    herb_name = urdf_module.SendCommand(args)
+    if sim:
+        urdf_uri = 'package://herb_description/robots/herb.urdf'
+        srdf_uri = 'package://herb_description/robots/herb.srdf'
+        args = 'Load {:s} {:s}'.format(urdf_uri, srdf_uri)
+        herb_name = urdf_module.SendCommand(args)
+    else:
+        import rospy
+        if not rospy.core.is_initialized():
+            raise RuntimeError('rospy not initialized. '
+                               'Must call rospy.init_node()')
+        urdf_string = rospy.get_param('/robot_description', None)
+        if urdf_string is None:
+            raise RuntimeError('rosparam "/robot_description" is not set.'
+                               ' Unable to load correct HERB model.')
+        srdf_string = rospy.get_param('/semantic_robot_description', None)
+        if srdf_string is None:
+            raise RuntimeError('rosparam "/semantic_robot_description" is not'
+                               ' set. Unable to load correct HERB model.')
+        urdf_json_wrapper = json.dumps(
+            {'urdf': urdf_string.replace('\n', ' ').replace('\r', ' '),
+             'srdf': srdf_string.replace('\n', ' ').replace('\r', ' ')})
```

Comment:
JSON requires escaped control characters,and rosparam returns with raw newlines (apparently) so I was getting parse errors. I could do the string replacement with the escaped newlines and returns, but spaces or nothing are fine for XML.